src/operator/channel_op_common.h

```diff
@@ -101,6 +101,42 @@ void Split(const mshadow::Tensor<xpu, dim, DType> &input,
     split_helper<xpu, dim, dim-1>(input, output, dimension, req);
   }
 }
+
+template<typename cpu, int dim, typename DType>
+void Split_2D(const mshadow::Tensor<cpu, dim, DType> &input,
+           std::vector<mshadow::Tensor<cpu, dim, DType> > *output,
+           const int dimension, const std::vector<OpReqType> &req) {
+  if (dimension != 1) {
+    LOG(FATAL) << "dimension (" << dimension << ") must == 1";
+  }
+  if (dim != 3) {
+    LOG(FATAL) << "dimension (" << dim << ") must == 3";
+  } else {
+    std::vector<mshadow::Tensor<cpu, dim, DType> > out = *output;
+    size_t size = out.size();
+    std::vector<int>slice_len;
+    std::vector<int>begin_pos;
+    begin_pos.push_back(0);
+
+    for (index_t i = 0; i < size; ++i) {
+      slice_len.push_back(out[i].size(dimension));
+      begin_pos.push_back(begin_pos[i] + out[i].size(dimension));
+    }
+#pragma omp parallel for num_threads(engine::OpenMP::Get()->GetRecommendedOMPThreadCount())
```

Comment:
Is it possible to parallel both 2 outer loops? Try collapse keyword. And beware that it's only works on Linux platform. You can find the example code in https://github.com/apache/incubator-mxnet/blob/master/src/operator/l2_normalization.cc